### PR TITLE
Feature/context menu

### PR DESCRIPTION
**Pull request by mschindlerMM**

### CODE REVIEW

File: ag-grid-webcomponent/index.js [hunk 1]
```diff
@@ -123,16 +125,48 @@ class AgGrid extends HTMLElement {
...
     this.api.setColumnDefs(defs);
   }
 
+  set getContextMenuItems(data) {
+    function prepareContextAction(item, params) {
+      if (typeof item === "string") return item;
+
+      if (typeof item.action === "string") {
+        let port = APP.ports[item.action];
+        item.action = () => port.send(params);
```
[the remark this answers is not in the document]
Comment (hingew) [reply ag-grid-webcomponent/index.js:145]: I agree with you. I will update this,  to use event communication :+1:

File: examples/src/Main.elm [hunk 2]
```diff
@@ -135,6 +143,13 @@ update msg model =
         ( NoOp, _ ) ->
             ( model, Cmd.none )
 
+        ( GotCounterIncremented _, _ ) ->
+            let
+                _ =
+                    Debug.log "increment counter" True
```
[the remark this answers is not in the document]
Comment (hingew) [reply examples/src/Main.elm:149]: Yea sounds good :) I will update this after updating to event communication.

File: ag-grid-webcomponent/index.js [hunk 3]
```diff
@@ -123,16 +125,55 @@ class AgGrid extends HTMLElement {
...
     this.api.setColumnDefs(defs);
   }
 
+  set getContextMenuItems(data) {
+    const prepareContextAction = (item, params) => {
+      if (typeof item === "string") return item;
+
+      if (typeof item.actionName === "string") {
+        item.action = () => {
+          const contextMenuEvent = new CustomEvent("oncontextaction", {
```
Comment (mschindlerMM) [inline ag-grid-webcomponent/index.js:145]: This event's name is a bit inconsistent with the others. There's no `on` part. In Elm, this reads a bit clunky: `Html.Events.on "oncontextaction" event`. Also, AgGrid events are like `cellValueChanged` or `gridReady` too.

Also, the other custom events were using camelCase. We might wanna stick with that.

File: ag-grid-webcomponent/index.js [hunk 4]
```diff
@@ -123,16 +125,55 @@ class AgGrid extends HTMLElement {
...
+            detail: {
+              action: item.actionName,
+              data: params.node.data,
+            }
+          });
+
+          console.log(params)
```
Comment (mschindlerMM) [inline ag-grid-webcomponent/index.js:152]: We should remove this `console.log`

File: ag-grid-webcomponent/index.js [hunk 5]
```diff
@@ -284,8 +325,9 @@ const setterProperties = Object.entries(
   .map(([key]) => key);
 
 export default class ElmAgGrid {
-  constructor({ apps = {}, aggregations = {} } = {}) {
+  constructor({ app = null, apps = {}, aggregations = {} } = {}) {
     window.ElmAgGridComponentRegistry = apps;
+    APP = app;
```
Comment (mschindlerMM) [inline ag-grid-webcomponent/index.js:330]: We can remove this now. We don't need the `app`

File: examples/src/Main.elm [hunk 6]
```diff
@@ -72,21 +77,24 @@ init flags url navKey =
...
-        Grouping _ ->
-            Sub.none
+            Grouping _ ->
+                Sub.none
 
-        RowSelection _ ->
-            Sub.none
+            RowSelection _ ->
+                Sub.none
+        , incrementCounter GotCounterIncremented
```
Comment (mschindlerMM) [inline examples/src/Main.elm:96]: This can be removed too

File: examples/src/Main.elm [hunk 7]
```diff
@@ -135,6 +143,13 @@ update msg model =
         ( NoOp, _ ) ->
             ( model, Cmd.none )
 
+        ( GotCounterIncremented _, _ ) ->
```
Comment (mschindlerMM) [inline examples/src/Main.elm:146]: This can be removed too

File: ag-grid-webcomponent/index.js (hunk 3 above)
Comment (hingew) [reply ag-grid-webcomponent/index.js:145]: Ok, what do you think about `contextActionClicked`?

Comment (mschindlerMM) [reply ag-grid-webcomponent/index.js:145]: Sounds good to me.

**Review by hingew — APPROVED** [review]

**Pull request merged**